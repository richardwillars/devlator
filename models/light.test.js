const lightModule = require('./light');

describe('models/light', () => {
  it('should export the schema', () => {
    const constants = {
      light: {
        a: 'A',
        b: 'B',
        c: 'C',
        d: 'D',
      },
    };
    const events = {
      a: 'A',
      b: 'B',
      c: 'C',
      d: 'D',
      e: 'E',
      f: 'F',
      g: 'G',
    };

    const mongoose = {
      model: () => {},
      Schema: class Schema {},
    };
    const lightModel = lightModule(mongoose, events, constants);
    expect(lightModel.schema).toMatchSnapshot();
  });

  it('should export the model', () => {
    const constants = {
      light: {
        a: 'A',
        b: 'B',
        c: 'C',
        d: 'D',
      },
    };
    const events = {
      a: 'A',
      b: 'B',
      c: 'C',
      d: 'D',
      e: 'E',
      f: 'F',
      g: 'G',
    };

    const modelMock = {};
    const mongoose = {
      model: jest.fn().mockReturnValue(modelMock),
      Schema: class Schema {
        constructor(props) {
          this.props = props;
        }
      },
    };
    const lightModel = lightModule(mongoose, events, constants);

    expect(lightModel.model).toEqual(modelMock);
    expect(mongoose.model).toHaveBeenCalledTimes(1);
    expect(mongoose.model.mock.calls[0][0]).toEqual('Light');
    const initialisedSchema = mongoose.model.mock.calls[0][1];
    expect(initialisedSchema instanceof mongoose.Schema).toEqual(true);
    expect(initialisedSchema.props).toMatchSnapshot();
  });
});
