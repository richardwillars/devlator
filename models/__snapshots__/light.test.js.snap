// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/light should export the model 1`] = `
Object {
  "_id": false,
  "additionalInfo": Object {
    "default": Object {},
    "required": false,
    "type": [Function],
  },
  "commands": Object {
    "breatheEffect": Object {
      "description": "Performs a breathe effect by fading between colours",
      "friendly": "Breathe effect",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "cycles": Object {
            "duration": "double",
            "maximum": 99999,
            "minimum": 0.01,
          },
          "fromColour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "peak": Object {
            "maximum": 1,
            "minimum": 0,
            "type": "double",
          },
          "period": Object {
            "maximum": 100,
            "minimum": 0.01,
            "type": "double",
          },
          "persist": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "colour",
          "period",
          "cycles",
          "persist",
          "peak",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "pulseEffect": Object {
      "description": "Performs a pulse effect by flashing between colours",
      "friendly": "Pulse effect",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "cycles": Object {
            "duration": "double",
            "maximum": 99999,
            "minimum": 0.01,
          },
          "fromColour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "period": Object {
            "maximum": 100,
            "minimum": 0.01,
            "type": "double",
          },
          "persist": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "colour",
          "period",
          "cycles",
          "persist",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "setBooleanState": Object {
      "description": "Turns the light on or off",
      "friendly": "Turn on or off",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "duration": Object {
            "maximum": 99999,
            "minimum": 0,
            "type": "integer",
          },
          "on": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "on",
          "duration",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "setBrightnessState": Object {
      "description": "Sets the brightness of the light",
      "friendly": "Set the brightness",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0.01,
                "type": "double",
              },
            },
            "required": Array [
              "brightness",
            ],
            "type": "object",
          },
          "duration": Object {
            "maximum": 99999,
            "minimum": 0,
            "type": "integer",
          },
        },
        "required": Array [
          "colour",
          "duration",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "setHSBState": Object {
      "description": "Sets the hue, saturation and brightness, allowing you to change the colour of the light",
      "friendly": "Set colour and brightness",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0.01,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "duration": Object {
            "maximum": 99999,
            "minimum": 0,
            "type": "integer",
          },
        },
        "required": Array [
          "colour",
          "duration",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "toggle": Object {
      "description": "Toggles the light on and off",
      "friendly": "Toggle",
      "type": [Function],
    },
  },
  "events": Object {
    "a": "A",
    "b": "B",
    "c": "C",
    "d": "D",
  },
  "name": Object {
    "required": true,
    "type": [Function],
  },
  "originalId": Object {
    "required": true,
    "type": [Function],
  },
}
`;

exports[`models/light should export the schema 1`] = `
Object {
  "_id": false,
  "additionalInfo": Object {
    "default": Object {},
    "required": false,
    "type": [Function],
  },
  "commands": Object {
    "breatheEffect": Object {
      "description": "Performs a breathe effect by fading between colours",
      "friendly": "Breathe effect",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "cycles": Object {
            "duration": "double",
            "maximum": 99999,
            "minimum": 0.01,
          },
          "fromColour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "peak": Object {
            "maximum": 1,
            "minimum": 0,
            "type": "double",
          },
          "period": Object {
            "maximum": 100,
            "minimum": 0.01,
            "type": "double",
          },
          "persist": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "colour",
          "period",
          "cycles",
          "persist",
          "peak",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "pulseEffect": Object {
      "description": "Performs a pulse effect by flashing between colours",
      "friendly": "Pulse effect",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "cycles": Object {
            "duration": "double",
            "maximum": 99999,
            "minimum": 0.01,
          },
          "fromColour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "period": Object {
            "maximum": 100,
            "minimum": 0.01,
            "type": "double",
          },
          "persist": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "colour",
          "period",
          "cycles",
          "persist",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "setBooleanState": Object {
      "description": "Turns the light on or off",
      "friendly": "Turn on or off",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "duration": Object {
            "maximum": 99999,
            "minimum": 0,
            "type": "integer",
          },
          "on": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "on",
          "duration",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "setBrightnessState": Object {
      "description": "Sets the brightness of the light",
      "friendly": "Set the brightness",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0.01,
                "type": "double",
              },
            },
            "required": Array [
              "brightness",
            ],
            "type": "object",
          },
          "duration": Object {
            "maximum": 99999,
            "minimum": 0,
            "type": "integer",
          },
        },
        "required": Array [
          "colour",
          "duration",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "setHSBState": Object {
      "description": "Sets the hue, saturation and brightness, allowing you to change the colour of the light",
      "friendly": "Set colour and brightness",
      "requestSchema": Object {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "properties": Object {
          "colour": Object {
            "properties": Object {
              "brightness": Object {
                "maximum": 1,
                "minimum": 0.01,
                "type": "double",
              },
              "hue": Object {
                "maximum": 360,
                "minimum": 0,
                "type": "integer",
              },
              "saturation": Object {
                "maximum": 1,
                "minimum": 0,
                "type": "double",
              },
            },
            "required": Array [
              "hue",
              "saturation",
              "brightness",
            ],
            "type": "object",
          },
          "duration": Object {
            "maximum": 99999,
            "minimum": 0,
            "type": "integer",
          },
        },
        "required": Array [
          "colour",
          "duration",
        ],
        "type": "object",
      },
      "type": [Function],
    },
    "toggle": Object {
      "description": "Toggles the light on and off",
      "friendly": "Toggle",
      "type": [Function],
    },
  },
  "events": Object {
    "a": "A",
    "b": "B",
    "c": "C",
    "d": "D",
  },
  "name": Object {
    "required": true,
    "type": [Function],
  },
  "originalId": Object {
    "required": true,
    "type": [Function],
  },
}
`;
